drivers/adc/adc_stm32.c

```diff
@@ -557,7 +557,7 @@ static int adc_stm32_init(struct device *dev)
 		LOG_ERR("ADC clock rate get error.");
 	}
 
-	wait_cycles = CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / adc_rate *
+	wait_cycles = SystemCoreClock / adc_rate *
```

Comment:
Why is this `SystemCoreClock` vs sys_clock_hw_cycles_per_sec()?

Reply:
Requested by code owner for this file some review rounds ago. Note, that there should be CPU frequency as the `wait_cycles` is used to spin around:
```
	for (int i = wait_cycles; i >= 0; i--) {
	}
```

Reply:
@erwango does this need to be `SystemCoreClock`?